tests/io/hdf5_test.py

```diff
@@ -299,6 +296,18 @@ def test_data_group():
     check_roundtrip(dg)
 
 
+def test_data_group_empty():
+    dg = sc.DataGroup()
+    check_roundtrip(dg)
+
+
+def test_data_group_unsupported_PyObject():
+    dg = sc.DataGroup({'a': x, 'b': sc.scalar([1, 2], dtype=object)})
```

Comment:
I meant a value that is not a scipp object but a plain Python int or dict.

Reply:
Ah, ok.
I went with logging a warning as this case is conceptually similar to an unsupported dtype.